.github/workflows/ci.yml

```diff
@@ -0,0 +1,127 @@
+name: ci
+on:
+  push:
+    branches:
+      - "**"
+
+defaults:
+  run:
+    # NOTE: A bit stricter than the default bash options used by GitHub Actions
+    # (bash --noprofile --norc -e -o pipefail {0})
+    shell: bash --noprofile --norc -euo pipefail {0}
+
+# NOTE: Set concurrency for the current workflow to 1
+concurrency: ci-${{ github.ref }}-${{ github.workflow }}
+
+jobs:
+  build-and-deploy:
+    timeout-minutes: 60
+    runs-on: ubuntu-22.04
+    permissions:
+      actions: read
+      contents: read
+      id-token: write
+    steps:
+      - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+
+      - uses: capralifecycle/actions-lib/check-runtime-dependencies@72f450a706d380212723f38753a82d8bb40ea5fc # v1
+
+      - uses: capralifecycle/actions-lib/parse-config@72f450a706d380212723f38753a82d8bb40ea5fc # v1
+        id: config
+        with:
+          config-file: ".ldp.json"
+
+      - uses: actions/setup-node@b39b52d1213e96004bfcb1c61a8a6fa8ab84f3e8 # v4.0.1
+        with:
+          node-version: 16.20.2
+
+      - uses: capralifecycle/actions-lib/configure-npm@72f450a706d380212723f38753a82d8bb40ea5fc # v1
+
+      - uses: capralifecycle/actions-lib/configure-aws-credentials@72f450a706d380212723f38753a82d8bb40ea5fc # v1
+        id: aws
+        with:
+          aws-account-id: ${{ steps.config.outputs.accountId }}
+          # NOTE: We use different roles on default and non-default branches
+          aws-iam-role-name: ${{ github.ref == format('refs/heads/{0}', github.event.repository.default_branch) && steps.config.outputs.roleName || steps.config.outputs.limitedRoleName }}
+
+      - name: install dependencies
+        run: npm ci
+
+      - name: build
+        run: npm run build
+
+      - name: lint
+        run: npm run lint
+
+      - name: test
+
+        run: npm run test
+
+      - name: verify cdk snapshots
+        working-directory: ./packages/infrastructure
+        run: npm run snapshots && git status && git add -N && git diff --exit-code
+
+      - name: create cloud assembly archive
+        working-directory: ./packages/infrastructure
+        if: ${{ github.ref_name == github.event.repository.default_branch }}
```

Comment:
Depending on the type of trigger, `github.ref_name` isn't always set to the name of the branch. I believe `github.ref == format('refs/heads/{0}', github.event.repository.default_branch)` should be more robust

Reply:
Alright, will update, wasn't easy to find an authoritative answer on this one